core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/ClusterBootstrapper.java

```diff
@@ -106,21 +103,117 @@
 
     public void bootstrapMachines(Long stackId) throws CloudbreakException {
         Stack stack = stackService.getByIdWithListsInTransaction(stackId);
-        String stackOrchestratorType = stack.getOrchestrator().getType();
-        OrchestratorType orchestratorType = orchestratorTypeResolver.resolveType(stackOrchestratorType);
+        bootstrapOnHost(stack);
+    }
 
-        if (orchestratorType.hostOrchestrator()) {
-            bootstrapOnHost(stack);
-        } else if (orchestratorType.containerOrchestrator()) {
-            LOGGER.debug("Skipping bootstrap of the machines because the stack's orchestrator type is '{}'.", stackOrchestratorType);
-        } else {
-            LOGGER.error("Orchestrator not found: {}", stackOrchestratorType);
-            throw new CloudbreakException("HostOrchestrator not found: " + stackOrchestratorType);
-        }
+    public void reBootstrapMachines(Long stackId) throws CloudbreakException {
+        Stack stack = stackService.getByIdWithListsInTransaction(stackId);
+        LOGGER.info("ReBootstrapMachines for stack [{}] [{}]", stack.getName(), stack.getResourceCrn());
+        reBootstrapOnHost(stack);
     }
 
     @SuppressFBWarnings("REC_CATCH_EXCEPTION")
     public void bootstrapOnHost(Stack stack) throws CloudbreakException {
+        Set<Node> nodes = collectNodesForBootstrap(stack);
+        try {
```

Comment:
bootstrap and reBootstrap are almost the same except for the save and update. Can this be refactored  to a common method?